### PR TITLE
Feature/53 login activity

### PR DESCRIPTION
**Pull request by ToufiPF**
The login activity, and its insertion into the app.
Users can access the login screen by clicking on the sign-in menu item ("se connecter") in the option menu I created for the occasion (accessible in HomeActivity only, at least for now).
The user can use the login screen to sign-in, using its email and password (via pindex) or alternatively click on a button to sign-in with google.
*Actual methods to sign-in/out are not implemented yet*
When an user is connected, the menu item to sign-in disappears and the one to sign-out appears instead.

Reviews appreciated !

### CODE REVIEW

File: app/src/main/java/com/pindex/main/data/PindexUser.kt [hunk 1]
```diff
@@ -1,15 +1,18 @@
...
  * Holds all data an user can have
  */
+@Parcelize
 data class PindexUser(
         val uid: String,
-        val name: String?,
-        val mail: String?,
-        val favorites: List<BlockDto>?,
+        val name: String? = null,
+        val mail: String? = null,
```
Comment (qtipee) [inline app/src/main/java/com/pindex/main/data/PindexUser.kt:14]: Shouldn't we use `email` instead of `mail`?

Comment (ToufiPF) [reply app/src/main/java/com/pindex/main/data/PindexUser.kt]: Indeed thanks for pointing out the typo

**Review by amomo25 — APPROVED** [review]
The simplification in ExperienceActivity was a great idea, the rest of the code is clear and well-documented.

**Review by lucasrlt — APPROVED** [review]
Good idea to use Mockito to mock the view model 
We still get 0% coverage on our repositories even with Mockito and Hilt (I have the same problem with locations), I don't know how we'll find a way to get past it

Really great job, it was not an easy one !

**Pull request merged**